icmplib/sockets.py

```diff
@@ -317,20 +317,22 @@ def receive(self, request=None, timeout=2):
         if not self._sock:
             raise SocketUnavailableError
 
-        self._sock.settimeout(timeout)
-        time_limit = time() + timeout
+        start_time = time()
+        time_limit = start_time + timeout
 
         try:
             while True:
+                remaining_time = time_limit - time()
+                if remaining_time <= 0:
+                    raise socket.timeout
+
+                self._sock.settimeout(remaining_time)
```

Comment:
```suggestion
                self._sock.settimeout(max(remaining_time, 0))
```

My suggestion to simplify logic and give the package one last chance to arrive even if the OS is heavily loaded.

Reply:
It's a good idea. I compared your suggestion with what the `async receive()` method is doing and I believe it would operate the same way you're suggesting, which is to read first and then let the OS enforces an immediate timeout.

My only concern would be tagging the packet with a timestamp that is over the timeout value, as that would look confusing to users. If they set a timeout of 500ms, they shouldn't ever expect a result with a value over 500ms even by the tiniest fraction. But since the async method is already potentially accepting ever-so-slightly-late packets, I'm OK to apply your suggestion and think about this separately.